### ISSUE TITLE
CloneOf field in generated romlist not populated

### ISSUE THREAD

**Issue by rsn8887** [mickelson/attract#162]
Thank you for the great frontend - is there a donation option?

I am using Attract Mode 1.6.0.

I have an xml file "fba_list.xml" in mame format, created by the emulator Finalburn Alpha. When using attract mode to create the romlist from this xml file, the cloneof field is blank for all entries!

The command used was
attract.exe -i fba_list.xml finalburn
where finalburn is an emulator configured in attract mode.

Example xml entry from fba_list.xml:
```xml
	<game name="19xxj" cloneof="19xx" romof="19xx">
		<description>19XX - the war against destiny (960104 Japan, yellow case)</description>
		<year>1996</year>
		<manufacturer>Capcom</manufacturer>
		<rom name="19xj-03b.6a" size="524288" crc="bcad93dd"/>
		<rom name="19xj-04b.7a" size="524288" crc="931882a1"/>
		<rom name="19xj-05b.8a" size="524288" crc="e7eeddc4"/>
		<rom name="19xj-06b.9a" size="524288" crc="f27cd6b8"/>
		<rom name="19xj-07.6d" size="524288" crc="61c0296c"/>
		<rom name="19x-69.4j" size="524288" crc="427aeb18"/>
		<rom name="19x-59.4d" size="524288" crc="63bdbf54"/>
		<rom name="19x-79.4m" size="524288" crc="2dfe18b5"/>
		<rom name="19x-89.4p" size="524288" crc="cbef9579"/>
		<rom name="19x-73.8j" size="524288" crc="8e81f595"/>
		<rom name="19x-74.9j" size="524288" crc="6d7ad22e"/>
		<rom name="19x-75.10j" size="524288" crc="cb1a1b6a"/>
		<rom name="19x-76.11j" size="524288" crc="26fc2b08"/>
		<rom name="19x-63.8d" size="524288" crc="6f8b045e"/>
		<rom name="19x-64.9d" size="524288" crc="ccd5725a"/>
		<rom name="19x-65.10d" size="524288" crc="6cf6db35"/>
		<rom name="19x-66.11d" size="524288" crc="16115dd3"/>
		<rom name="19x-83.8m" size="524288" crc="c11f88c1"/>
		<rom name="19x-84.9m" size="524288" crc="68cc9cd8"/>
		<rom name="19x-85.10m" size="524288" crc="f213666b"/>
		<rom name="19x-86.11m" size="524288" crc="574e0473"/>
		<rom name="19x-93.8p" size="524288" crc="9fad3c55"/>
		<rom name="19x-94.9p" size="524288" crc="e10e252c"/>
		<rom name="19x-95.10p" size="524288" crc="2b86fa67"/>
		<rom name="19x-96.11p" size="524288" crc="ae6eb692"/>
		<rom name="19x-01.1a" size="131072" crc="ef55195e"/>
		<rom name="19x-51.6a" size="524288" crc="e9cd7780"/>
		<rom name="19x-52.7a" size="524288" crc="b27b91a8"/>
		<rom name="19x-53.8a" size="524288" crc="2e563ee2"/>
		<rom name="19x-54.9a" size="524288" crc="f47c1f24"/>
		<rom name="19x-55.10a" size="524288" crc="0b1af6e0"/>
		<rom name="19x-56.11a" size="524288" crc="dfa8819f"/>
		<rom name="19x-57.12a" size="524288" crc="229ba777"/>
		<rom name="19x-58.13a" size="524288" crc="c7dceba4"/>
	</game>
```
Corresponding romlist line created by attract mode: 
```csv
19xxj;19XX - the war against destiny (960104 Japan, yellow case);finalburn;;1996;Capcom;Shooter / Flying Vertical;2P sim;;;;;;;;
```
Note the empty CloneOf field after "finalburn" and before the year "1996"! This makes filtering out the clones impossible. 

EDIT: typos fixed.

**Comment by rsn8887** [comment]
Now it crashes when I try to generate my romlist based on the mame-format xml file generated by finalburn alpha. To reproduce, use the file truncated fba_list_xml (content below) and the command:
```
attract.exe --import-romlist fba_list_truncated.xml finalburn
```

The crash also happens if I add the xml filename in the gui (after nplayers.ini filename).

```xml
<?xml version="1.0"?>
<!DOCTYPE datafile PUBLIC "-//FB Alpha//DTD ROM Management Datafile//EN" "http://www.logiqx.com/Dats/datafile.dtd">

<datafile>
	<header>
		<name>FB Alpha - Arcade Games</name>
		<description>FB Alpha v0.2.97.37 Arcade Games</description>
		<category>Standard DatFile</category>
		<version>0.2.97.37</version>
		<author>FB Alpha</author>
		<homepage>http://www.fbalpha.com/</homepage>
		<url>http://www.fbalpha.com/</url>
		<clrmamepro forcenodump="ignore"/>
	</header>
	<game name="99lstwar" cloneof="repulse" romof="repulse">
		<description>'99: The Last War</description>
		<year>1985</year>
		<manufacturer>Proma</manufacturer>
		<rom name="1999.4f" size="8192" crc="e3cfc09f"/>
		<rom name="1999.4h" size="8192" crc="fd58c6e1"/>
		<rom name="7.j4" merge="7.j4" size="8192" crc="57a8e900"/>
		<rom name="1.f2" merge="1.f2" size="8192" crc="c485c621"/>
		<rom name="2.h2" merge="2.h2" size="8192" crc="b3c6a886"/>
		<rom name="3.j2" merge="3.j2" size="8192" crc="197e314c"/>
		<rom name="repulse.b4" merge="repulse.b4" size="8192" crc="86b267f3"/>
		<rom name="1999.4a" size="4096" crc="49a2383e"/>
		<rom name="15.9h" merge="15.9h" size="8192" crc="c9213469"/>
		<rom name="16.10h" merge="16.10h" size="8192" crc="7de5d39e"/>
		<rom name="17.11h" merge="17.11h" size="8192" crc="0ba5f72c"/>
		<rom name="8.6a" merge="8.6a" size="16384" crc="0e9f757e"/>
		<rom name="9.7a" merge="9.7a" size="16384" crc="f7d2e650"/>
		<rom name="10.8a" merge="10.8a" size="16384" crc="e717baf4"/>
		<rom name="11.9a" merge="11.9a" size="16384" crc="04b2250b"/>
		<rom name="12.10a" merge="12.10a" size="16384" crc="d110e140"/>
		<rom name="13.11a" merge="13.11a" size="16384" crc="8fdc713c"/>
		<rom name="b.1j" merge="b.1j" size="256" crc="3ea35431"/>
		<rom name="g.1h" merge="g.1h" size="256" crc="acd7a69e"/>
		<rom name="r.1f" merge="r.1f" size="256" crc="b7f48b41"/>
		<rom name="m1.2c" merge="m1.2c" size="32" crc="83a39201"/>
	</game>
</datafile>
```

**Comment by mickelson** [comment]
Hey there,

The file you provided seems to work just fine on linux (debian jessie).

What version of windows are you using?

**Comment by rsn8887** [comment]
Using Windows 10 it just crashes here. 

Exact steps to reproduce on my machines (tested with Windows 10 desktop and Windows 10 running in VMWare on my macbook pro):
- extract attached zip into attract mode directory (creates files text.xml and 99lstwar.zip (empty)).
- start attract mode
- create emulator, name "finalburn"
- change rompath to "./"
- change additional files to "./test.xml"
- click generate romlist
- it crashed with the windows requester showing.
[xmltest.zip](https://github.com/mickelson/attract/files/65633/xmltest.zip)

**Comment by rsn8887** [comment]
I am not exactly sure when this was fixed, but the error is fixed in version 2.0.0 release!
